File: Source/Aladdin.CAPI.Java/src/aladdin/capi/Authentication.java
package aladdin.capi;
import java.io.*; 

///////////////////////////////////////////////////////////////////////////
// Аутентификация
///////////////////////////////////////////////////////////////////////////
public abstract class Authentication
{
    // имя пользователя
    public abstract String user(); 
    
    // тип аутентификации
    public abstract Class<? extends Credentials>[] types(); 

    // выполнить аутентификацию
    public abstract Credentials[] authenticate(SecurityObject obj) throws IOException; 
}

File: Source/Aladdin.CAPI/CAPI/Authentication.cs
using System; 

namespace Aladdin.CAPI
{
	///////////////////////////////////////////////////////////////////////////
    // Протокол аутентификации
	///////////////////////////////////////////////////////////////////////////
    public abstract class Authentication
    {
        // тип пользователя и тип аутентификации
        public abstract string User { get; } public abstract Type[] Types { get; }

        // выполнить аутентификацию
        public abstract Credentials[] Authenticate(SecurityObject obj); 
    }
}
